pkg/kubelet/volumemanager/volume_manager.go

```diff
@@ -357,9 +357,10 @@ func (vm *volumeManager) WaitForAttachAndMount(pod *v1.Pod) error {
 		}
 
 		return fmt.Errorf(
-			"timeout expired waiting for volumes to attach/mount for pod %q/%q. list of unattached/unmounted volumes=%v",
+			"timeout expired waiting for volumes to attach/mount for pod %q/%q with error %v. list of unattached/unmounted volumes=%v",
```

Comment:
IMO, `err` can be only `ErrWaitTimeout` here. And that's already logged.

Did you check that real flex error gets surfaced here? Can you post sample error string?

Reply:
yeah I doubt this will help surfacing the error because all the `verifyMountedVolumes`  func again does is to count umounted volumes. So this  will not help surface the actual flex error I think.

Reply:
Right. The `conditionFunc` (`vm.verifyVolumesMountedFunc`) will not return error. Then `wait.Poll` will only return timeout error.